src/shellingham/posix/__init__.py

```diff
@@ -71,6 +71,11 @@ def _get_shell(cmd, *args):
     if cmd.startswith("-"):  # Login shell! Let's use this.
         return _get_login_shell(cmd)
     name = os.path.basename(cmd).lower()
+    if name == "rosetta" or name.contains("qemu-"):
```

Comment:
For the qemu part, what does a typical executable name look like? I’m mostly thinking whether the check can be made more conservative, e.g. using a prefix check (instead of `contains`, which matches anywhere in the name) or some sort of regex.

Reply:
You're absolutely right this should be a regex based on the binaries listed in the [qemu docs](https://www.qemu.org/docs/master/user/main.html)